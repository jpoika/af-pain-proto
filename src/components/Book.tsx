import * as React from "react";
import { Link } from 'react-router';
import BookCover from  './BookCover';
import Page from './Page';
import Paper from 'material-ui/Paper';
import Menu from 'material-ui/Menu';
import MenuItem from 'material-ui/MenuItem';
import Helmet from 'react-helmet';

export interface Props { 
  appBarTitle(msg: string): any;
  book: any;
  isOpen: boolean;
  pageOpen(any): void;
  title: string;
  //currentPage: number;
}

export interface State {
  numPages: number;
  currentPage: number;
  isOpen: boolean;
}

// 'HelloProps' describes the shape of props.
// State is never set so we use the 'undefined' type.
const pageMenu = (pages,pageClick) => {
  return (
      <Menu>
       {pages.map((page,i) => <MenuItem key={page.id} primaryText={page.title} onTouchTap={pageClick(i)} />)}
      </Menu>
    )
}

export default class Book extends React.Component<Props, State> {
    constructor(props){
      super(props);

      this.state = {
        numPages: props.book.pages.length || 0,
        currentPage: 0,
        isOpen: this.props.isOpen || false
      }
    }
    componentWillMount () {
      const {title} = this.props;
      this.props.appBarTitle(title);
    }

    componentWillReceiveProps(nextProps) {
      const {book,isOpen,title} = nextProps;

      this.props.appBarTitle(title);
      this.setState({isOpen});
    }

    handlePageClick = (pageIdx) => {
      const {pageOpen} = this.props;
      return (event) => {
        this.setState({currentPage: pageIdx, isOpen: true});
        pageOpen(event);
      }
    }

    handleBookClose = (event) => {
      this.setState({isOpen: false});
    }

    handlePageNext = (event) => {
      let nextPage = this.state.currentPage + 1;
      if(nextPage < this.state.numPages){
        window.scrollTo(0,0);
        this.setState({currentPage: nextPage});
      }
    }

    handlePagePrevious = (event) => {
      let prevPage = this.state.currentPage - 1;
      if(prevPage > -1){
        window.scrollTo(0,0);
        this.setState({currentPage: prevPage});
      }
    }


    render() {
      const {book} = this.props;
        const page = book.pages[this.state.currentPage] ? book.pages[this.state.currentPage] : book.pages[0];
        const currentPageContent = <Page next={this.handlePageNext} previous={this.handlePagePrevious} close={this.handleBookClose} pageIndex={this.state.currentPage} numPages={this.state.numPages} page={page} />;
        let content = currentPageContent;
        if(!this.state.isOpen){
          content = pageMenu(book.pages,this.handlePageClick);
          return (<div>
                      <Helmet>
                          <title>{book.title}</title>
                      </Helmet>
                      <BookCover book={book} />
                      
                        {content}
                  
                  </div>
                );
        } else {
          return currentPageContent
        }
    }
}