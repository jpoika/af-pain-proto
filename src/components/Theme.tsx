import * as React from 'react';
import * as ReactDOM from 'react-dom';
import SplashPage from './SplashPage';

import getMuiTheme from 'material-ui/styles/getMuiTheme';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import siteTheme from './customTheme';

const muiTheme = getMuiTheme(siteTheme);
interface MyProps {
  [propName: string]: any;
}

interface MyState {
  [propName: string]: any;
}


class Theme extends React.Component<MyProps,  MyState>{

  render () {
    return (
      <MuiThemeProvider muiTheme={getMuiTheme(muiTheme)}>
              {this.props.children || <SplashPage />}
              {/* SplashPage is displayed if there are no children */}

      </MuiThemeProvider>
    );
  }
}

export default Theme;