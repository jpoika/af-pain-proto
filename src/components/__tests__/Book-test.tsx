import * as React from 'react';
import * as ReactDOM from 'react-dom';
import Theme from '../Theme';
import Book from '../Book';
import BookPage from '../Page';
import { shallow } from 'enzyme';

import * as renderer from 'react-test-renderer';


const appBarTitleCb = () => {

}
const pageOpenCb= () => {

}
const BookItem = {
  title: "The Catcher in the Rye",
  pages: [
    {
      id: 1,
      title: 'chapter 1',
      content: "blah blah 1"
    },
    {
      id: 2,
      title: 'chapter 2',
      content: "blah blah 2"
    },
    {
      id: 3,
      title: 'chapter 3',
      content: "blah blah 3"
    }
  ]
}



test('Test NotFound Snapshot', () => {

  const component = renderer.create(
    <Theme>
        <Book appBarTitle={appBarTitleCb} pageOpen={pageOpenCb} book={BookItem} isOpen={true} title={"Page Title"} />
    </Theme>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});

test('Test Enzyme Open Book Has Page', () => {

  const wrapper = shallow(
        <Book appBarTitle={appBarTitleCb} pageOpen={pageOpenCb} book={BookItem} isOpen={true} title={"Page Title"} />
  );
  expect(wrapper.find(BookPage).length).toBe(1);

});

test('Test Enzyme Closed Book Has No Page', () => {

  const wrapper = shallow(
        <Book appBarTitle={appBarTitleCb} pageOpen={pageOpenCb} book={BookItem} isOpen={false} title={"Page Title"} />
  );
  expect(wrapper.find(BookPage).length).toBe(0);

});

