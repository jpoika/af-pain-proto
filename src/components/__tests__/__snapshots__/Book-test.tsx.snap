// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test NotFound Snapshot 1`] = `
<div>
  <div
    style={
      Object {
        "display": "flex",
        "flexFlow": "row nowrap",
        "justifyContent": "space-between",
      }
    }
  >
    <div
      style={
        Object {
          "flex": "0 0",
          "marginTop": "10px",
        }
      }
    >
      <button
        disabled={true}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOut={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "background": "none",
            "border": 10,
            "boxSizing": "border-box",
            "cursor": "default",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 0,
            "fontWeight": "inherit",
            "height": 48,
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "visible",
            "padding": 12,
            "position": "relative",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
            "width": 48,
            "zIndex": 1,
          }
        }
        tabIndex={-1}
        type="button"
      >
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(33, 33, 33, 0.3)",
              "cursor": "not-allowed",
              "display": "inline-block",
              "fill": "rgba(33, 33, 33, 0.3)",
              "height": 24,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": 24,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </svg>
      </button>
    </div>
    <div
      style={
        Object {
          "flex": "0 0",
          "marginTop": "10px",
        }
      }
    >
      <button
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOut={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "background": "none",
            "border": 10,
            "boxSizing": "border-box",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 0,
            "fontWeight": "inherit",
            "height": 48,
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "visible",
            "padding": 12,
            "position": "relative",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
            "width": 48,
            "zIndex": 1,
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <svg
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "#212121",
                "display": "inline-block",
                "fill": "currentColor",
                "height": 24,
                "muiPrepared": true,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "width": 24,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </div>
      </button>
    </div>
    <div
      style={
        Object {
          "flex": "0 0",
          "marginTop": "10px",
        }
      }
    >
      <button
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOut={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "background": "none",
            "border": 10,
            "boxSizing": "border-box",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 0,
            "fontWeight": "inherit",
            "height": 48,
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "visible",
            "padding": 12,
            "position": "relative",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
            "width": 48,
            "zIndex": 1,
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <svg
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "#212121",
                "display": "inline-block",
                "fill": "currentColor",
                "height": 24,
                "muiPrepared": true,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "width": 24,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
            />
          </svg>
        </div>
      </button>
    </div>
  </div>
  <h2>
    chapter 1
  </h2>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "blah blah 1",
      }
    }
  />
  <div
    style={
      Object {
        "display": "flex",
        "flexFlow": "row nowrap",
        "justifyContent": "space-between",
      }
    }
  >
    <div
      style={
        Object {
          "flex": "0 0",
          "marginTop": "10px",
        }
      }
    >
      <button
        disabled={true}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOut={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "background": "none",
            "border": 10,
            "boxSizing": "border-box",
            "cursor": "default",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 0,
            "fontWeight": "inherit",
            "height": 48,
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "visible",
            "padding": 12,
            "position": "relative",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
            "width": 48,
            "zIndex": 1,
          }
        }
        tabIndex={-1}
        type="button"
      >
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(33, 33, 33, 0.3)",
              "cursor": "not-allowed",
              "display": "inline-block",
              "fill": "rgba(33, 33, 33, 0.3)",
              "height": 24,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": 24,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </svg>
      </button>
    </div>
    <div
      style={
        Object {
          "flex": "0 0",
          "marginTop": "10px",
        }
      }
    >
      <button
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOut={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "background": "none",
            "border": 10,
            "boxSizing": "border-box",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 0,
            "fontWeight": "inherit",
            "height": 48,
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "visible",
            "padding": 12,
            "position": "relative",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
            "width": 48,
            "zIndex": 1,
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <svg
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "#212121",
                "display": "inline-block",
                "fill": "currentColor",
                "height": 24,
                "muiPrepared": true,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "width": 24,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </div>
      </button>
    </div>
    <div
      style={
        Object {
          "flex": "0 0",
          "marginTop": "10px",
        }
      }
    >
      <button
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOut={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "background": "none",
            "border": 10,
            "boxSizing": "border-box",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 0,
            "fontWeight": "inherit",
            "height": 48,
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "visible",
            "padding": 12,
            "position": "relative",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
            "width": 48,
            "zIndex": 1,
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <svg
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "#212121",
                "display": "inline-block",
                "fill": "currentColor",
                "height": 24,
                "muiPrepared": true,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "width": 24,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
            />
          </svg>
        </div>
      </button>
    </div>
  </div>
</div>
`;
